examples/siro_sandbox/sandbox_app.py

```diff
@@ -214,19 +228,23 @@ def get_play_episodes_ids(play_episodes_filter_str):
         return dataset
 
     def _env_step(self, action):
-        self._obs = self.env.step(action)
-        self._metrics = self.env.get_metrics()
+        (
+            self._obs,
+            reward,
+            done,
+            self._metrics,
+        ) = self.gym_habitat_env.direct_hab_step(action)
```

Comment:
Why can't you use the already-public `step`? Why do you have to also expose `direct_hab_step`?

Reply:
@ASzot I'd appreciate your thought here. 

We may have one or two controllers at SandboxApp runtime:
**Single agent case**
1. SingleAgentBaselineController - one policy-controlled agent
2. GuiController - one GUI-controlled agent

**Multi agent case**
1. MultiAgentBaselineController - multiple policy-controlled agents
2. [GuiController, SingleAgentBaselineController] - one GUI-controlled agent + one policy-controlled agent
3. [GuiController, MultiAgentBaselineController] - one GUI-controlled agent + multiple policy-controlled agents (**not supported yet**)

In **Multi agent case**, for example, option 2: observation and action spaces are split between two controllers and both GuiController and SingleAgentBaselineController return an action. Currently, both of these actions are already in the form {"action": ..., "action_args": ...} i.e. for GuiController the action is constructed in that form and for SingleAgentBaselineController `continuous_vector_action_to_hab_dict` is called after `agent.actor_critic.act()`.

After that, these actions are concatenated: 
```python
  all_names = []
  all_args = {}
  for controller in self.controllers:
      ctrl_action = controller.act(obs, self._env)
      all_names.extend(ctrl_action["action"])
      all_args.update(ctrl_action["action_args"])
  action = {"action": tuple(all_names), "action_args": all_args}
```

Then I want to pass aggregated action to the environment (HabGymWrapper). But it's step method doesn't accept actions in the form {"action": ..., "action_args": ...}. To sidestep this I exposed `direct_hab_step` and call it directly.

How would you suggest to combine controllers actions and pass to the environment? Would it be better to have controllers returning actions as `np.array`, then merging them and passing to the env.step (not having controllers returning {"action": ..., "action_args": ...})?

Reply:
I suggest having the controllers always returning actions as `np.array` and calling `.step` instead of `.direct_hab_step`. The Gym API is the interface for interacting with the Habitat environment. You should refrain from using the underlying Habitat internals such as the `{"action": ..., "action_args": ...}` action format.

Reply:
Is there a reason for this, @ASzot ? I find it hard to get the right index in the np.array containing all the actions, and dict seems more intuitive. Is there a check that the gymwrapper is doing that we should account for?

In the previous version of the sandbox, we were calling the `habitat.core.env.Env` in the sandbox. Why not keeping that one instead of the `GymWrapper`?